### ISSUE TITLE
VW average loss is wrong when pdrop is used

### ISSUE THREAD

**Issue by marco-rossi29** [VowpalWabbit/vowpal_wabbit#2076]
#### Describe the bug
VW doesn't take into consideration pdrop in the computation of average loss

#### To Reproduce
Data file used (pdrop = 0.5)
```
>cat 1line.json
{"_label_cost":-1,"_label_probability":0.8,"_label_Action":1,"_labelIndex":0,"Timestamp":"2019-08-11T00:00:00.0010000Z","Version":"1","EventId":"337bc167","a":[1,2],"c":{"s":{"e":"1"},"_multi":[{"ns":{"f":2.6}},{"ns":{"f":1.6}}]},"p":[0.8,0.2],"VWState":{"m":"e90dd08d-3272-42cd-afd6-6d9d0294923d/e8481e06-b505-4997-bab8-f46069a3bb74"},"pdrop":0.5}
```

VW output
```
>vw --dsjson -d 1line.json --cb_adf -P 1
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = 1line.json
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
-1.250000 -1.250000            1            1.0    known        0:0...        4

finished run
number of examples = 1
weighted example sum = 1.000000
weighted label sum = 0.000000
average loss = -1.250000
total feature number = 6
```

#### Observed Behavior
Average Loss is simply: _label_cost/_label_probability = -1/0.8 = -1.25

#### Expected behavior
Average Loss should be: _label_cost/_label_probability/pdrop = -1/0.8/0.5 = -2.5

#### Environment
What version of VW did you use? 8.7.0 (tested with current latest master d05599e7071 and with commit d5ba0788c)
What OS or language did you use? Windows command line

**Comment by jackgerrits** [comment]
I'd say this is something that should be done when the dsjson format is generated and logged. VW only uses _label_probability, I don't think VW itself should be enlightened to pdrop

**Comment by ataymano** [comment]
Looks like we have regressions on this.

**Comment by marco-rossi29** [comment]
Let's get to it. How can I help?

Do you know at what point the regression was introduced?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ataymano** [comment]
There are actually 2 problems here:
1) regression which is causing us ignoring pdrop
2) bug in initializing original weights - they are initialized as (1 - pdrop) while it should be 1/(1-pdrop)
Will send PR shortly and we can discuss there.

**Comment by marco-rossi29** [comment]
Sounds great. Thank you for tackle it immediately!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by ataymano** [comment]
Actually looks like this bug is invalid and we should close it and work on #3056. @marco-rossi29, what do you think?
